apps/firebase-api/tests/users.courses.submissions.spec.ts

```diff
@@ -0,0 +1,178 @@
+import * as firebase from '@firebase/rules-unit-testing';
+
+import {
+  PROJECT_ID,
+  getAuthedFirestore,
+  updateUser,
+  getUserRef,
+} from './utils';
+
+import { getUserSubmissionRef } from './utils';
+
+const ALICE_ID = 'alice';
+const BOB_ID = 'bob';
+const DAN_ID = 'dan';
+const RANDOM_SUBMISSION_ID = 'random_review';
+const COURSE_A = 'course_a';
+const COURSE_B = 'course_b';
+
+describe('users/{{userID}}/courses/{{courseId}}/submissions/{{submissionId}}', () => {
+  beforeEach(async () => {
+    await firebase.clearFirestoreData({ projectId: PROJECT_ID });
+  });
+
+  it('create', async () => {
+    // bob is mentor of course_a
+    await updateUser(BOB_ID, {
+      is_mentor: true,
+      mentorable_courses: [COURSE_A],
+    });
+
+    // alice can create
+    const aliceDb = getAuthedFirestore({ uid: ALICE_ID });
+    await firebase.assertSucceeds(
+      getUserSubmissionRef(aliceDb, ALICE_ID, COURSE_A, RANDOM_SUBMISSION_ID).set({ new: true })
+    );
+
+    // others cannot create
+    const anyoneDb = getAuthedFirestore(null);
+    await firebase.assertFails(
+      getUserSubmissionRef(anyoneDb, ALICE_ID, COURSE_A, RANDOM_SUBMISSION_ID).set({ new: false })
+    );
+
+    // bob cannot create
+    const bobDb = getAuthedFirestore({ uid: BOB_ID });
+    await firebase.assertFails(
+      getUserSubmissionRef(bobDb, ALICE_ID, COURSE_A, RANDOM_SUBMISSION_ID).set({ new: false })
+    );
+  });
+
+  it('read: mentors or owner can read', async () => {
+    // bob is mentor of course_a
+    await updateUser(BOB_ID, {
+      is_mentor: true,
+      mentorable_courses: [COURSE_A],
+    });
+    // dan is mentor of course_b
+    await updateUser(DAN_ID, {
+      is_mentor: true,
+      mentorable_courses: [COURSE_B],
+    });
+
+    // alice can get
+    const aliceDb = getAuthedFirestore({ uid: ALICE_ID });
+    await firebase.assertSucceeds(
+      getUserSubmissionRef(aliceDb, ALICE_ID, COURSE_A, RANDOM_SUBMISSION_ID).set({
+        new: true,
+        mentor: getUserRef(aliceDb, BOB_ID),
+      })
+    );
+    await firebase.assertSucceeds(
+      getUserSubmissionRef(aliceDb, ALICE_ID, COURSE_A, RANDOM_SUBMISSION_ID).get()
+    );
+
+    // others cannot get
+    const anyoneDb = getAuthedFirestore(null);
+    await firebase.assertFails(
+      getUserSubmissionRef(anyoneDb, ALICE_ID, COURSE_A, RANDOM_SUBMISSION_ID).get()
+    );
+
+    // bob can get: since he is mentor of RANDOM_SUBMISSION_ID
+    const bobDb = getAuthedFirestore({ uid: BOB_ID });
+    await firebase.assertSucceeds(
```

Comment:
I think there's been a huge misunderstanding here. Nowhere in this document is Bob assigned as the mentor of this submission. You assigned him to be able to review the course - but you never assigned him to be the mentor assigned to that submission. You should go through the code related to the mentor dashboard and see what I mean here. I have a feeling that this whole file is understanding the data model incorrectly.

Could we maybe try this PR again? I'm happy to sit down with you to explain the entire process in detail. :)